### DISCUSSION TITLE
Sync command replaces main language source files

### DISCUSSION THREAD

**Discussion by drumsta** [Alexandre-Fernandez/astro-i18n#24] [Q&A]
Hello, Alexandre!

I'm giving a try for your i18n solution for Astro. However, I must be doing something silly as I run `astro-i18n sync` command, it replaces the content of source files for the main language as well.

Original source files:
![image](https://user-images.githubusercontent.com/1282771/221961423-ccfa28e1-47a4-4791-8934-4f81736163f0.png)

After I run `astro-i18n sync` the source files look as below:
![image](https://user-images.githubusercontent.com/1282771/221961636-2db30af0-f443-4b12-8cbd-f2aa08c881e1.png)

- Additional folders for defined languages have been created as expected
- However original source files have been amended as well.

My config files are pretty standard:
![image](https://user-images.githubusercontent.com/1282771/221962517-965ef385-665f-438b-b07a-ed2d0c519582.png)

Any ideas what I'm missing?

If you need access to the source files, you can find the repo (`dev` branch) at https://github.com/drumsta/companion-for-ts/tree/dev

**Comment by Alexandre-Fernandez** [comment]
https://github.com/Alexandre-Fernandez/astro-i18n#supportedlangcodes
https://github.com/Alexandre-Fernandez/astro-i18n/discussions/19

**Comment by drumsta** [comment]
Thanks for pointing it out, it works now!

Apparently, it was a copy-paste issue migrating settings from another i18n solution used previously.

**Comment by drumsta** [comment]
Hello, Alexandre!

I don't want create another thread, so I will ask here.

The proxy pages created by `astro-i18n sync` command have an issue related to the backslash in the `import` directive, i.e. it should be a forward slash instead of backslash.

I'm using VS Code on Windows 11.

Is there any settings to manage this issue?

![image](https://user-images.githubusercontent.com/1282771/222462159-6a911ac5-9a2b-4a71-b7eb-a8325d024d9c.png)

**Comment by Alexandre-Fernandez** [comment]
Hi, I don't have a windows machine, try the command with this package and show me the output.
[astro-i18n-1.6.6.zip](https://github.com/Alexandre-Fernandez/astro-i18n/files/10872688/astro-i18n-1.6.6.zip)
You can also try to fix it yourself the code that generates the import path is here : 
https://github.com/Alexandre-Fernandez/astro-i18n/blob/2eaca61b939bd22cc6ddb33459f38e760a2d0b26/src/core/fs/generators/page.proxy.ts

**Comment by drumsta** [comment]
Here you go...

![image](https://user-images.githubusercontent.com/1282771/222475005-fa0a5c3b-ea7d-4bfa-be05-e8cdfb5d6bb8.png)

I'll try to take a look at your code later and let you know.

**Comment by Alexandre-Fernandez** [comment]
Isn't it working ? your proxy page looks like it does have forward slash import and the output you get does show backslash transformed to forward ones : `\[...slug].astro` => `"../[...slug].astro"`.
The package I gave you is just the current version with some console logs.
